layers/core/layer.py

```diff
@@ -0,0 +1,95 @@
+import json
+from layers.core.exceptions import UninitializedLayer, BadType, BadInput, \
+    handler
+from layers.core.layerobj import _LayerObj
+
+
+class Layer:
+    def __init__(self, init_dict={}, strict=True):
+        """
+             Initialization - create a new Layer object
+             :param init_dict: Optionally provide base Layer json or string
+                data on initialization
+         """
+        self.__layer = None
+        self.strict = strict
+        self.load_input(init_dict)
+
+    @property
+    def layer(self):
+        if self.__layer is not None:
+            return self.__layer
+        return "No Layer Loaded Yet!"
+
+    def load_input(self, init_dict):
+        """
+            loads input from a string or dictionary format
+            :param init_dit: the string or dictionary representing the layer
+                data to be loaded
+        """
+        if isinstance(init_dict, str):
+            self.data = json.loads(init_dict)
```

Comment:
shouldn't layer.data have a `_` prefix since it's an internal value? I often use `dir` to remind myself of the API:

```
dir(l1)
['_Layer__layer', '__class__', '__delattr__', '__dict__', '__dir__', '__doc__', '__eq__', '__format__', '__ge__', '__getattribute__', '__gt__', '__hash__', '__init__', '__init_subclass__', '__le__', '__lt__', '__module__', '__ne__', '__new__', '__reduce__', '__reduce_ex__', '__repr__', '__setattr__', '__sizeof__', '__str__', '__subclasshook__', '__weakref__', '_build', 'data', 'export_file', 'get_dict', 'layer', 'load_file', 'load_input', 'strict']
```

Without the prefix I'm led to believe that directly accessing `data` is the best way to get at the internal layer data when it really should be `to_dict`/`get_dict`.